k2/python/k2/autograd.py

```diff
@@ -186,6 +182,40 @@ def backward(
         return None, None, None, None, None, None, grad_a, grad_b
 
 
+class _IndexSelectFunction(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, src: torch.Tensor, index: torch.Tensor) -> torch.Tensor:
+        '''Returns a new tensor which indexes the input tensor along dimension 0
+        using the entries in `index`.
+
+        If the entry in `index` is -1, then the corresponding entry in the
+        returned tensor is 0.
+
+        Caution:
+          `index.dtype == torch.int32` and `index.ndim == 1`.
+
+        Args:
+          src:
+            The input tensor. Either 1-D or 2-D with dtype torch.int32 or torch.float32.
+          index:
+            1-D tensor of dtype torch.int32 containing the indexes to index. If an entry
```

Comment:
Just say "containing the indexes", not "indexes to index".  And you can add the sentence at the end: "The values in `index` should be in the range [-1..src.shape[0]-1].

Reply:
Fixed.